annotator/static/canvas.js

```diff
@@ -163,26 +167,10 @@ class Canvas {
             myState.selection = newBox;
             myState.enlargeDirection = Box.border.BOTTOMRIGHT;
 
-            var thingDom = document.createElement("li");
-            thingDom.className = "list-group-item col-xs-6";
-            thingDom.id = newThing.fill;
-            var length =  newThing.type.length + 70;
-            thingDom.style = "color: azure; background-color: " + newThing.fill + "; width: " + length + "px";
-            thingDom.innerText = newThing.type;
-            thingDom.addEventListener("click", function() {
-                myState.selection = myState.getBox(newThing);
-                if (!myState.selection) {
-                    var firstBox = newThing.keyframes[0];
-                    var previousBox = new Box(newThing, myState.frame, firstBox.x, firstBox.y, firstBox.w, firstBox.h);
-
-                    myState.boxes.push(previousBox);
-                    myState.selection = previousBox;
-                }
-                myState.valid = false;
-            });
+            newThing.drawBox(myState);
```

Comment:
Sorry, by box I meant the html button. That's only updated when a new thing is created. Redrawing every box is too costly in my opinion since it modifies the dom.